bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -130,27 +132,100 @@ public void run()
         storageHelperExecutor.shutdown();
         try {
           serverHelperExecutor.awaitTermination(5000, TimeUnit.MILLISECONDS);
+          storageHelperExecutor.awaitTermination(5000, TimeUnit.MILLISECONDS);
         } catch (InterruptedException ex) {
           logger.debug("Executor Termination", ex);
         }
-        logger.info("Server stopped listening at {}", address);
+        logger.info("Server stopped listening at {}", getServerAddress());
+        latch.countDown();
+        latch = null;
```

Comment:
Typically CountDownLatch is used in scenarios where you have more than one threads. It feels like an overkill for this scenario.